### PR TITLE
NATIVE-139: Fix loading of categories and change FetcherModule

### PR DESCRIPTION
**Pull request by maxammann**

### CODE REVIEW

**Comment by maxammann** [comment]
Failing CI is fixed in NATIVE-140

File: src/modules/endpoint/sagas/fetchResourceCache.js [hunk 1]
```diff
@@ -29,13 +33,16 @@ export default function * fetchResourceCache (city: string, language: string, fe
...
 
     if (Platform.OS !== 'android') {
-      return {failureMessages: {}, fetchedUrls: {}}
+      return {}
     }
 
-    const result: FetchResultType = yield call(new FetcherModule().fetchAsync, targetUrls, progress => {})
+    const results: FetchResultType = yield call(new FetcherModule().fetchAsync, targetUrls, progress => {})
+
+    const successResult = pickBy(results, result => !result.errorMessage)
```
Comment (Schedulaar) [inline src/modules/endpoint/sagas/fetchResourceCache.js]: successResult*s*

File: src/modules/endpoint/ResourceURLFinder.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  _foundUrls: Set<string> = new Set<string>()
+
+  init () {
+    this.onAttributeTagFound = (name: string, value: string) => {
+      if (name === 'href' || name === 'src') {
+        if (['png', 'jpg', 'jpeg', 'pdf'].includes(getExtension(value))) {
+          this.foundUrls.add(value)
+        }
+      }
+    }
```
Comment (Schedulaar) [inline src/modules/endpoint/ResourceURLFinder.js]: Why is this function definition part of init?
also foundUrls should probably be emptied (or initialized) here...

File: src/modules/endpoint/sagas/loadCityContent.js [hunk 3]
```diff
@@ -34,12 +34,8 @@ export default function * loadCityContent (database: MemoryDatabase, newCity: st
...
 
-  console.dir(database.categoriesMap)
-
-  // yield call(() => database.writeCategories())
-  // yield call(() => database.writeResourceCache())
-  // yield call(() => database.readCategories())
-  // yield call(() => database.readResourceCache())
-
-  console.dir(database.categoriesMap)
+  //yield call(() => database.writeCategories())
```
Comment (Schedulaar) [inline src/modules/endpoint/sagas/loadCityContent.js]: Add a comment or remove

File: src/modules/fetcher/FetcherModule.js [hunk 4]
```diff
@@ -5,12 +5,7 @@ import NativeFetcherModule from './NativeFetcherModule'
 
 export type TargetFilePathsType = {[path: string]: string}
 
-export type FailureMessageType = { [url: string]: string }
-
-export type FetchResultType = {
-  failureMessages: FailureMessageType,
-  fetchedUrls: { [path: string]: { lastUpdate: string, url: string} }
-}
+export type FetchResultType = { [path: string]: { lastUpdate: string, url: string, errorMessage: string} }
```
Comment (Schedulaar) [inline src/modules/fetcher/FetcherModule.js]: missing exact object type:
`{ [path: string]: {| lastUpdate: string, url: string, errorMessage: string |} }`

File: src/modules/categories/components/CategoryListItem.js [hunk 5]
```diff
@@ -41,9 +41,10 @@ class CategoryListItem extends React.Component<PropsType> {
   renderSubCategories (): Array<React.Node> {
     const {subCategories} = this.props
     return subCategories.map(subCategory =>
-      <SubCategoryListItem key={subCategory.id}
-                           subCategory={subCategory} onItemPress={this.props.onItemPress}
-                           theme={this.props.theme} />
+      <View key={subCategory.path}>
```
Comment (NKlug) [inline src/modules/categories/components/CategoryListItem.js]: Why do we need the `View` as a wrapper here?

File: src/modules/categories/components/CategoryListItem.js [hunk 6]
```diff
@@ -41,9 +41,10 @@ class CategoryListItem extends React.Component<PropsType> {
...
+        <SubCategoryListItem key={subCategory.path} subCategory={subCategory} onItemPress={this.props.onItemPress}
+                             theme={this.props.theme} />
```
[the remark this answers is not in the document]
Comment (maxammann) [reply src/modules/categories/components/CategoryListItem.js]: What do you mean with destructor?

File: src/modules/endpoint/sagas/fetchResourceCache.js (hunk 1 above)
Comment (maxammann) [reply src/modules/endpoint/sagas/fetchResourceCache.js]: The type has changed to a single map. Error results and Sucess results are now in one map.

File: src/modules/endpoint/sagas/loadCityContent.js (hunk 3 above)
Comment (maxammann) [reply src/modules/endpoint/sagas/loadCityContent.js]: Done in NATIVE-140

File: src/modules/categories/components/CategoryListItem.js (hunk 5 above)
Comment (maxammann) [reply src/modules/categories/components/CategoryListItem.js]: Done

File: src/modules/fetcher/FetcherModule.js (hunk 4 above)
Comment (maxammann) [reply src/modules/fetcher/FetcherModule.js]: Doone

File: src/modules/endpoint/ResourceURLFinder.js (hunk 2 above)
Comment (maxammann) [reply src/modules/endpoint/ResourceURLFinder.js]: You can run findResourceUrls multiple times. Each time the foundUrls is cleared. There is no problem with creating that function in the init() as we only need to create it once.

Comment (Schedulaar) [reply src/modules/endpoint/ResourceURLFinder.js]: do it :D

File: src/modules/categories/components/CategoryListItem.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (Schedulaar) [reply src/modules/categories/components/CategoryListItem.js]: do it :D

**Pull request merged**